core/metadata/src/model/ProtoDescriptorHelper.ts

```
@@ -0,0 +1,203 @@
+    }
+    // 2. access level match
+    if (proto.accessLevel !== AccessLevel.PUBLIC && proto.instanceModuleName !== ctx.moduleName) {
+      return false;
+    }
+    // 3. qualifier match
+    if (!QualifierUtil.matchQualifiers(proto.qualifiers, ctx.qualifiers)) {
+      return false;
+    }
+    return true;
+  }
+}
```

Biome

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)